_posts/2025-02-25-dev-traffic-proxy-mit-proxychains-ng.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Development Traffic Proxy mit ProxyChains-NG
+author: sfast
+categories: [Quick Tips]
+tags: [tools, dev, proxychains]
+shortdesc: Leite den gesamten Traffic einer Anwendung mit proxychains-ng durch einen Proxy.
+---
+
+**Schneller testen, weniger deployen**. 
+Anstatt auf die Pipeline zu warten, einfach `proxychains-ng` verwenden, um lokale Anwendungen direkt mit entfernten Services zu verbinden.
+Mit `proxychains-ng` lässt sich der gesamte Traffic eines Programms über einen Proxy umleiten.
+
+Anhand eines Kubernetes-Clusters wird demonstriert, wie lokale Anwendungen auf Services im Cluster zugreifen können – ohne Warten auf die Pipeline und aufwändiges Deployment.
+
+## Einführung
+
+Beim Entwickeln von Anwendungen, die auf externe Systeme wie APIs, Datenbanken oder Message Queues (beispielsweise Kafka) zugreifen, sind lokale Tests oft problematisch. 
+`kubectl port-forward` hilft in vielen Fällen, aber nicht immer – insbesondere, wenn Dienste zusätzliche DNS-Namen nutzen, die lokal nicht auflösbar sind.
+
+Hier kommt **`proxychains-ng`** ins Spiel. 
+Es leitet den gesamten Traffic einer lokalen Anwendung durch einen Proxy im Cluster und ermöglicht so den direkten Zugriff auf entfernte Services.
+
+Ein einfaches Beispiel zeigt, wie ein `curl`-Befehl direkt auf einen Service im Kubernetes-Cluster zugreift. 
+Dieselbe Technik funktioniert auch für komplexere Szenarien: 
+Eine lokal gestartete Java-Anwendung mit einem Kafka-Consumer kann sich nahtlos mit einem Kafka-Broker im Cluster verbinden und dort Nachrichten konsumieren – ohne jedes Mal ein Docker-Image bauen und deployen zu müssen.
+Selbst wenn ein Broker über Service Discovery zusätzliche DNS-Namen bereitstellt, die normalerweise nur innerhalb des Clusters auflösbar sind, werden auch diese Verbindungen über `proxychains-ng` nahtlos umgeleitet.
+
+## Setup
+
+### Installation der Tools
+
+- [`proxychains-ng`](https://github.com/rofl0r/proxychains-ng)
+  - `brew install proxychains-ng`
+  - Leitet den Traffic von lokalen Anwendungen durch einen Proxy.
+  - Nach der Installation ist `proxychains4` als Befehl verfügbar.
+- [`shadowsocks-rust`](https://github.com/shadowsocks/shadowsocks-rust)
+  - `brew install shadowsocks-rust`
+  - Ein schneller und sicherer Proxy-Server und -Client.
+
+**Optional**:
+- [`k3d`](https://k3d.io/)
+  - `brew install k3d`
+  - Zum Erstellen eines lokalen Kubernetes-Clusters, um die Beispiele auszuprobieren.
+
+### Konfiguration von `proxychains-ng`
+
+Nach der Installation von `proxychains-ng` muss folgende Konfigurationsdatei angelegt werden:
+
+Dateipfad: `~/.proxychains/proxychains.conf`
+```conf
+dynamic_chain
+proxy_dns
+tcp_read_time_out 15000
+tcp_connect_time_out 8000
+
+[ProxyList]
+socks5  127.0.0.1 1080
+```
+- **`dynamic_chain`**: Verwendet die Proxys in der angegebenen Reihenfolge.
+- **`proxy_dns`**: Sorgt dafür, dass DNS-Anfragen ebenfalls durch den Proxy gehen.
+- **Timeouts:** Die TCP-Timeouts sind in Millisekunden angegeben.
```

Comment:
```suggestion
- **dynamic_chain**: Verwendet die Proxys in der angegebenen Reihenfolge.
- **proxy_dns**: Sorgt dafür, dass DNS-Anfragen ebenfalls durch den Proxy gehen.
- **Timeouts:** Die TCP-Timeouts sind in Millisekunden angegeben.
```

Ich würde die Formatierung anpassen. Verstehe warum du das so gemacht hast, sieht aber irgendwie broken aus:
![image](https://github.com/user-attachments/assets/a5a7863d-f3ba-4916-8bd9-d4d1bb6e4c1f)

It’s up to you.

Reply:
Ich hab's jetzt mal angepasst, sodass es einheitlicher aussieht:
<img width="736" alt="Anpassung" src="https://github.com/user-attachments/assets/17b7ed24-7334-4d6e-9f1c-11105aaa9c4c" />